### PR TITLE
Add fgdbg::Session.h, Pass.h, Resource.h dtos/models

### PR DESCRIPTION
**Pull request by feresr**
Still in WIP, feel free to take a look and leave suggestions

### CODE REVIEW

File: libs/fgdbg/include/fgdbg/Resource.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FGDBG_RESOURCE_H
+#define FGDBG_RESOURCE_H
+
+#include <string>
+
+namespace filament::fgdbg {
```
Comment (bejado) [inline libs/fgdbg/include/fgdbg/Resource.h:22]: Resource.h and Pass.h could probably be consolidated to a single header, unless you're planning on adding a lot more to them.

File: filament/src/details/Engine.cpp [hunk 2]
```diff
@@ -260,7 +260,8 @@ void FEngine::init() {
 
 #if FILAMENT_ENABLE_FGDBG
     if (!debug.fgdbg) {
-        debug.fgdbg = std::make_unique<fgdbg::DebugServer>();
+        auto server = std::shared_ptr<fgdbg::DebugServer>();
+        debug.fgdbg = std::make_unique<fgdbg::Session>("sample", server);
```
Comment (feresr) [inline filament/src/details/Engine.cpp:264]: `Server` is a shared resource among multiple `Sessions`
so it gets injected in the `Session` constructor

File: libs/fgdbg/include/fgdbg/Resource.h (hunk 1 above)
Comment (feresr) [reply libs/fgdbg/include/fgdbg/Resource.h:22]: I think that's a good idea! What do we usually call the enclosing .h file `models.h`?

File: filament/src/details/Engine.cpp [hunk 3]
```diff
@@ -652,7 +653,7 @@ int FEngine::loop() {
             static int fgdbgMessage = 0;
             fgdbgMessage++;
             if (fgdbgMessage % 100 == 0) {
-                debug.fgdbg->sendMessage(fgdbgMessage/100);
```
Comment (feresr) [inline filament/src/details/Engine.cpp:655]: `sendMessage()` is now private to `server`

**Review by pixelflinger — APPROVED** [review]

**Pull request merged**